File: frontend/src/services/stats.ts
import api from "./api";

export type StatsResponse = {
  poolHashrate: string;
  miners: number;
  workers: number;
  networkHeight: number;
  difficulty: string;
  poolLuck: number;
  poolFee: number;
  price: number;
};

export async function getStats(): Promise<StatsResponse> {
  const { data } = await api.get("/v1/dashboard");

  return {
    poolHashrate: data.network.poolHashrate.toString(),
    miners: data.workers.online,
    workers: data.workers.online,
    networkHeight: data.network.height,
    difficulty: data.network.difficulty.toString(),
    poolLuck: data.round.luck,
    poolFee: 1.5,
    price: 0,
  };
}

File: backend/internal/models/stats.go
package models

type Stats struct {
	PoolHashrate  string  `json:"poolHashrate"`
	Miners        int     `json:"miners"`
	Workers       int     `json:"workers"`
	NetworkHeight int     `json:"networkHeight"`
	Difficulty    string  `json:"difficulty"`
	PoolLuck      float64 `json:"poolLuck"`
	PoolFee       float64 `json:"poolFee"`
	Price         float64 `json:"price"`
}
